listenbrainz/webserver/static/js/src/recommendations/RecommendationCard.test.tsx

```diff
@@ -128,22 +140,23 @@ describe("submitFeedback", () => {
       />
     );
     const instance = wrapper.instance();
+    const updateFeedbackSpy = jest.spyOn(instance.props, "updateFeedback");
 
     const spy = jest.spyOn(instance.APIService, "submitRecommendationFeedback");
     spy.mockImplementation(() => Promise.resolve(200));
 
     instance.submitFeedback("love");
     expect(spy).toHaveBeenCalledTimes(0);
-    expect(instance.props.updateFeedback).toHaveBeenCalledTimes(0);
+    expect(updateFeedbackSpy).toHaveBeenCalledTimes(0);
     expect(wrapper.exists(".recommendation-controls")).toEqual(false);
   });
 
   it("doesn't call updateFeedback if status code is not 200", async () => {
     const wrapper = shallow<RecommendationCard>(
-      <RecommendationCard {...props} />
+      <RecommendationCard {...{ ...props, updateFeedback: jest.fn() }}/>
     );
     const instance = wrapper.instance();
-    props.updateFeedback = jest.fn();
+    const updateFeedbackSpy = jest.spyOn(instance.props, "updateFeedback");
```

Comment:
This only needs one jest function:
```javascript
<RecommendationCard {...props} />
…
const updateFeedbackSpy = jest.spyOn(instance.props, "updateFeedback");
```

There's no need to replace updateFeedback before spying on it.

Reply:
I think we need it

```
 static/js/src/recommendations/RecommendationCard.test.tsx (84.149s)
  ● submitFeedback › calls API, calls updateFeedback correctly

    TypeError: Cannot assign to read only property 'updateFeedback' of object '[object Object]'

      82 |     );
      83 |     const instance = wrapper.instance();
    > 84 |     const updateFeedbackSpy = jest.spyOn(instance.props, "updateFeedback");
         |                                    ^
      85 | 
      86 |     const spy = jest.spyOn(instance.APIService, "submitRecommendationFeedback");
      87 |     spy.mockImplementation(() => Promise.resolve(200));

      at ModuleMockerClass.spyOn (node_modules/jest-mock/build/index.js:800:26)
      at static/js/src/recommendations/RecommendationCard.test.tsx:84:36
      at step (static/js/src/recommendations/RecommendationCard.test.tsx:44:23)
      at Object.next (static/js/src/recommendations/RecommendationCard.test.tsx:25:53)
      at static/js/src/recommendations/RecommendationCard.test.tsx:19:71
      at Object.<anonymous>.__awaiter (static/js/src/recommendations/RecommendationCard.test.tsx:15:12)
      at Object.<anonymous> (static/js/src/recommendations/RecommendationCard.test.tsx:79:51)
```

Reply:
You're absolutely right.
`read only property 'updateFeedback'` Hm, that's very surprising to me, but hey.

After all, we define `updateFeedback` at the beginning of the file to be a mock function, so it doesn't matter.
When I commented last I thought we were replacing an actual functional method and could possibly interfere with the tests, but that's not the case.